include/psa/crypto_builtin_key_derivation.h

```diff
@@ -105,7 +105,8 @@ typedef struct psa_tls12_prf_key_derivation_s {
 } psa_tls12_prf_key_derivation_t;
 #endif /* MBEDTLS_PSA_BUILTIN_ALG_TLS12_PRF) ||
         * MBEDTLS_PSA_BUILTIN_ALG_TLS12_PSK_TO_MS */
-#if defined(MBEDTLS_PSA_BUILTIN_ALG_PBKDF2_HMAC)
+#if defined(MBEDTLS_PSA_BUILTIN_ALG_PBKDF2_HMAC) || \
```

Comment:
What I'm wondering in general now is if PBKDF2 deverves a constant by itself so that we don't have to put a concatenation of 'defined' conditions for stuff that applies to all the PBKDF2 related algorithms. This would also apply, for example, in `include/psa/crypto_driver_contexts_key_derivation.h`.
@gilles-peskine-arm does what I'm saying make sense?

Reply:
Yes, that makes sense, like we have `PSA_HAVE_SOFT_BLOCK_MODE` for generic block cipher handling code, we could have a `PSA_HAVE_SOFT_PBKDF2` for generic PBKDF2 code.